### ISSUE TITLE
FranticRain, does this mean we will never be able to connect xbox one controller wirelessly with the driver properly on Mac Big Sur like we can on OS X?!

### ISSUE THREAD

**Issue by Archit1134** [360Controller/360Controller#1110]
Correction: Kexts need to be re-compiled at least to work on Big Sur. Since we haven't been able to compile on the latest version of Xcode for years, it's not viable to upgrade the kext to Big Sur. Instead, I'll be focusing on DriverKit support. That work is in-progress.

_Originally posted by @FranticRain in https://github.com/360Controller/360Controller/issues/1084#issuecomment-648305679_

I posted few hours ago that I couldn't connect my xbox one controller to my iMac which runs Mac Big Sur as the controller always vibrated but never really connected (The light on the xbox one logo didn't turn on, in the controller).
I used to have Catalina and it worked perfectly. 
I do know that apple has made huge changes to Big Sur but we do know that Big Sur will soon be the OS everyone will be using when the stable version gets out, moving from Catalina and the previous OS X versions. So in the future does it mean we will never be able to connect xbox one controller to Macs? I think we would work on the driver to work on Big Sur. With respect, I think there is no point on developing this driver that only works with Catalina (Hope, for now) as soon we all are already looking forward to move to Big Sur when the stable gets out as the OS is bring new features and major improvements, Big Sur is the next era as OS X has ended and 11 has started. I think what we really need is a driver that makes xbox one controllers connect wired via micro-USB cable to Mac Big Sur. Please do remember that Catalina will become the OS of the past as soon as Big Sur stable is out. 
Please do the needful.

**Comment by FranticRain** [comment]
Hi Archit,

Unfortunately, I am not longer in a position to contribute to open source. Given that knowledge, it's clear that the kext isn't going to work on Big Sur, and that trying to get a controller kext working on Big Sur is a poor use of your time. I am also unable to contribute any sort of DriverKit code to replace it. I know for a fact that DriverKit will work as a viable replacement for 360Controller, and anyone is welcome to take up the mantle and build a dext to replace 360Controller.

However saying that you will "never be able to connect xbox one controller wirelessly with the driver properly on Mac Big Sur," is incredibly inaccurate. You can, right now, connect an Xbox One controller wirelessly to your Mac running Big Sur with Bluetooth. If your controller doesn't support Bluetooth, sorry. Additionally, "never" is a strong word since anyone could conceivably take my place and develop a dext to work with older Xbox One controllers. But I think it would behoove you get either a newer Xbox One controller that supports Bluetooth or (potentially) a new Xbox "core" controller. I assume that they use the same Bluetooth protocol so it should "just work," but I don't know that for sure.

So in conclusion: No. I won't do the needful. I have no interest in working for free anymore. A means exists that is supported by Apple, which eliminates the need for this driver. There is no reason to complicate the ecosystem with a dext or kext that may or may not be installed. Developers can simply now code against the CGController spec and interact with all (common) controllers equally. That's better for developers and it's better for end users (in terms of support).
